editor.planx.uk/src/pages/FlowEditor/components/Team/tests/TeamMembers.addNewEditor.test.tsx

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable jest/expect-expect */
+import { fireEvent, screen, within } from "@testing-library/react";
```

Comment:
Wherever possible, we're trying to use the `setup()` API for consistency. See an example here -

https://github.com/theopensystemslab/planx-new/blob/main/editor.planx.uk/src/@planx/components/AddressInput/Public.test.tsx#L10-L26

Docs: 
- https://testing-library.com/docs/user-event/setup/
- https://testing-library.com/docs/user-event/intro/#writing-tests-with-userevent

Reply:
Ah ok, thanks for explaining! 

I've updated to be able to get `user` from `setup` (and thus got rid of `fireEvent` but is there a particular reason why we prefer use `getBy...` functions from here rather than from `screen`?

I had a go at getting everything from `setup` but in my opinion it made the tests a lot more unwieldy due to having to initialise the variables globally (`rendered`) and not being able to destructure them so nicely. 

I also think it made it make less sense to have a helper called `setupTeamMembersScreen` obscuring some of the methods that were exposed. I ended up with this which I don't think is as nice:

```js
describe("when a user with the ADD_NEW_EDITOR feature flag enabled presses 'add a new editor'", () => {
  let rendered: Record<"user", UserEvent> & RenderResult

  beforeEach(async () => {
    rendered = setup(
      <DndProvider backend={HTML5Backend}>
        <TeamMembers teamMembersByRole={exampleTeamMembersData} />
      </DndProvider>
    );
    await rendered.findByText("Team editors"); 
    const teamEditorsTable = rendered.getByTestId("team-editors");
    const addEditorButton = await within(teamEditorsTable).findByText(
      "Add a new editor"
    );
    rendered.user.click(addEditorButton);
  });
  it("opens the modal and displays the input fields", async () => {
    expect(await rendered.findByTestId("modal-create-user-button")).toBeVisible();
    expect(await rendered.findByLabelText("First name")).toBeVisible();
  });
});
```

In the example you linked, `screen` is used as well as `user` so hopefully the way it is left now is acceptable 😅  Super interested in talking through the thought processes behind the current code conventions though so let me know if you'd like a chat!